src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/NewAzureRmAnalysisServicesServer.cs

```diff
@@ -0,0 +1,99 @@
+        public string Name { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 2, Mandatory = true,
+            HelpMessage = "Azure region where the server should be created.")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("North Central US", "South Central US", "Central US", "West Europe", "North Europe", "West US",
+            "East US",
+            "East US 2", "Japan East", "Japan West", "Brazil South", "Southeast Asia", "East Asia", "Australia East",
+            "Australia Southeast", IgnoreCase = true)]
+        public string Location { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 3, Mandatory = true,
+            HelpMessage =
+                "Name of the Sku used to create the server"
+            )]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("S1", "S2", "S4", "D1")]
+        public string Sku { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 4, Mandatory = false,
+            HelpMessage = "A string,string dictionary of tags associated with this server")]
+        [ValidateNotNull]
+        public Hashtable Tags { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 4, Mandatory = false,
+            HelpMessage = "A comma separated server names to set as administrators on the server")]
+        [ValidateNotNull]
+        public string Administrators { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(Name, Resources.CreateNewAnalysisServicesServer))
+            {
+                try
+                {
+                    if (AnalysisServicesClient.GetServer(ResourceGroupName, Name) != null)
+                    {
+                        throw new CloudException(string.Format(Resources.AnalysisServerExists, Name));
+                    }
+                }
+                catch (CloudException ex)
+                {
+                    if (ex.Body != null && !string.IsNullOrEmpty(ex.Body.Code) && ex.Body.Code == "ResourceNotFound" ||
+                        ex.Message.Contains("ResourceNotFound"))
+                    {
+                        // server does not exists so go ahead and create one
+                    }
+                    else if (ex.Body != null && !string.IsNullOrEmpty(ex.Body.Code) &&
+                             ex.Body.Code == "ResourceGroupNotFound" || ex.Message.Contains("ResourceGroupNotFound"))
+                    {
+                        // resource group not found, let create throw error don't throw from here
+                    }
+                    else
+                    {
+                        // all other exceptions should be thrown
+                        throw;
+                    }
+                }
+
+                WriteObject(AnalysisServicesClient.CreateOrUpdateServer(ResourceGroupName, Name, Location, Sku, Tags, Administrators));
```

Comment:
ou need to create PS display types for the types returned from your client that are the output of cmdlets - this will allow you to avoid breaking the PowerShell API when your management API changes